File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while (!s.equals("1")){
            int sizeBefore = s.length();
            s = s.replace("0", "");
            int sizeAfter = s.length();
            
            s = Integer.toBinaryString(sizeAfter);
            
            answer[1] += sizeBefore - sizeAfter;
            answer[0] += 1;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.kt
class Solution {
    fun solution(s: String): IntArray {
        var change: Int = 0
        var count: Int = 0
        var string: String = s
        
        while (string != "1"){
            change += 1
            count += string.replace("1", "").count()
            string = Integer.toBinaryString(string.replace("0", "").count())
        }
        
        return intArrayOf(change, count)
    }
}